tests/network/persistentips.go

```diff
@@ -0,0 +1,158 @@
+				var err error
+				vmi, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				return nil
+			}
+			currentVMIUD = func() (types.UID, error) {
+				if err := refreshVMI(); err != nil {
+					return "", err
+				}
+				return vmi.UID, nil
+			}
+
+			currentIPAMClaims = func() ([]ipamclaimsv1alpha1.IPAMClaim, error) {
+				ipamClaimList, err := kubevirt.Client().IPAMClaimsClient().K8sV1alpha1().IPAMClaims(vm.Namespace).List(context.Background(), metav1.ListOptions{})
+				if err != nil {
+					return nil, err
+				}
+				return ipamClaimList.Items, err
+			}
+		)
+		BeforeEach(func() {
+			Expect(checks.HasFeature(virtconfig.PersistentIPsGate)).To(BeTrue())
+
+			nad = &nadv1.NetworkAttachmentDefinition{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "persistentips",
+					Namespace: testsuite.GetTestNamespace(nil),
+				},
+				Spec: nadv1.NetworkAttachmentDefinitionSpec{
+					Config: fmt.Sprintf(`
+{
+        "cniVersion": "0.3.0",
+        "name": "persistentips",
+        "type": "ovn-k8s-cni-overlay",
+        "topology": "layer2",
+        "subnets": "10.100.200.0/24",
+        "netAttachDefName": "%s/persistentips",
+        "allowPersistentIPs": true
+}
+`, testsuite.GetTestNamespace(nil)),
+				},
+			}
+			By("Create NetworkAttachmentDefinition")
+			_, err := kubevirt.Client().NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(testsuite.GetTestNamespace(nil)).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			vm = libvmi.NewVirtualMachine(libvmifact.NewAlpine(
+				libvmi.WithNamespace(testsuite.GetTestNamespace(nil)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding("multus")),
+				libvmi.WithNetwork(libvmi.MultusNetwork("multus", "persistentips")),
+			), libvmi.WithRunning())
+
+			By("Creating VM using the nad")
+			vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Create(context.Background(), vm, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Wait for VMI to be created")
+			Eventually(refreshVMI).
+				WithPolling(time.Second).
+				WithTimeout(2 * time.Minute).
+				Should(Succeed())
+
+			By("Wait for VMI to be ready")
+			vmi = libwait.WaitUntilVMIReady(vmi, console.LoginToAlpine)
+
+			By("Check IPAMClaim get created")
+			ipamClaims, err := kubevirt.Client().IPAMClaimsClient().K8sV1alpha1().IPAMClaims(vm.Namespace).List(context.Background(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaims.Items).ToNot(BeEmpty())
+
+		})
+
+		JustAfterEach(func() {
+			By("Check no ipamclaims leftovers after VM delete")
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, metav1.DeleteOptions{})).To(Succeed())
+			Eventually(currentIPAMClaims).
+				WithTimeout(time.Minute).
+				WithPolling(time.Second).
+				Should(BeEmpty())
```

Comment:
there is a tear down code that runs AfterEach test no need

Reply:
This is part of the test, it checks that after deleting the VM there is no leftovers, it's not related to kubevirt e2e global tear down.

Reply:
ic
btw do we really need it ? we depend on the GC, and dont need to check it maybe
it can be added only once we delete ourselves when there is hotplug, but whatever you prefer

Reply:
We are checking that we are checking one of the functionalities and this is that IPAMClaim is delete we don't care if we implement it by GC or manually.